grunt-tasks/jst.js

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function(grunt) {
+  grunt.config("jst", {
+    compile: {
+      options: {
+        namespace: "wcTmpl",
+        prettify: true,
+        processContent: function(src) {
+          // strip the opening and closing <script> tags...
+          // otherwise, the template functions will just inject script elements
+          // that won't render.
+          src = src.split(/\<script type=['"]text\/template['"]>/)[1];
```

Comment:
@karlcow lol. :)

(below might be useful for @zoepage to understand as well)

Basically 99% of this PR is "pre-compiling" the JS templates and sticking them in a `templates.js` file, so they don't have to be parsed and compiled on page load. 

For example, the following template, in this PR is now in its own file called `state-button.jst`. 

```
<script type="text/template">
  <span class="js-comment-<%= stateClass %>"><%= state %></span>
</script>
```

Before, or rather, currently on master, the same template lived in the HTML file inline:

https://github.com/webcompat/webcompat.com/blob/0f56b8a7aa4cfda06fa7b0fe08367e015638d5c5/webcompat/templates/issue/issue-comment-submit.html#L39-L41

The `jst` command does the pre-compiling, and produces output that looks like this, which lives in the templates.js file:

```js
this["wcTmpl"]["issue/state-button.jst"] = function(obj) {obj || (obj = {});var __t, __p = '', __e = _.escape;with (obj) {__p += '<span class="js-comment-' +((__t = ( stateClass )) == null ? '' : __t) +'">' +((__t = ( state )) == null ? '' : __t) +'</span>';}return __p};
```

So instead of parsing the client-template when we load the page and try to render by passing it data, we just pass this function the data. This also eliminates the need to allow `eval` (or `new Function('blahblahblah')`, which is the same thing, which is a potential XSS attack vector.

The potential downside of this approach is we can't rely on jinja parsing interactions inside the JS templates anymore, since the JS templates are converted into JS before the page is even served.

So there's a few benefits:

1. security -- the real motivation for this PR is to remove `unsafe-eval` from CSP
2. performance -- no more client-side parsing of JS templates, and now we get to rely on `templates.js` living in the HTTP cache.

A few downsides:

1. You need to remember to re-build if you mess with templates, that didn't exist before.
2. We can't mix jinja and JS templates, which was kind of powerful (if not maybe confusing)

As for my original question about using string.split vs regexp... I'm just gonna use regexp since the shape of the opening and closing script tags is constant in the new templates (they don't need ids).